homeassistant/auth.py

```diff
@@ -491,6 +505,21 @@ def __init__(self, hass):
         await self.async_save()
         return client
 
+    async def async_get_or_create_client(self, name, redirect_uris, no_secret):
+        """Find a client, if not exists, create a new one."""
+        if self.clients is None:
+            await self.async_load()
+
+        redirect_uris_counter = Counter(redirect_uris)
+
+        for _, client in self.clients.items():
+            if (client.name == name
```

Comment:
I think that we should just match on name and not allow clients with duplicate names. That can only confuse users.